shm/StockManagement.hpp

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include "Cargo.hpp"
+
+class StockManagement {
+private:
+    static CargoPtr findCargo(const CargoStock& stock_, const CargoPtr& cargo);
+
+public:
+    static void removeCargoFromStock(CargoStock& stock_, const CargoPtr& cargo, size_t amount);
+    static void addCargoToStock(CargoStock& stock_, const CargoPtr& cargo, size_t amount);
```

Comment:
Probably it is a mistake from my temporary branch, but please change `stock_` name to `stock`. The same in StockManagement.cpp file.